docs/DeveloperGuide.md

```diff
@@ -257,329 +257,368 @@ _{Explain here how the data archiving feature will be implemented}_
+   Use case ends.
 
-**Use case: Delete a person**
+**Extension**
 
-**MSS**
+* 1a. TAB detects that the entered student data does not satisfy requirements
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+    * 1a1. TAB displays an error message and the recommended changes
 
-    Use case ends.
+      Use case resumes at step 2.
 
-**Extensions**
+* 1b. TAB detects a duplicate student
 
-* 2a. The list is empty.
+    * 1b1. TAB displays an error message
 
-  Use case ends.
+      Use case ends.
 
-* 3a. The given index is invalid.
+<br/>
 
-    * 3a1. AddressBook shows an error message.
-      
-  Use case resumes at step 2.
+#### Use case: UC3 – Edit a Student
 
-**Use case: UC3 Track Student's Lesson Information**
+**MSS:**
 
-**MSS**
+1. User [<ins>gets a list of students (UC1)</ins>](Use-case:-UC1---Get-a-list-of-students)
+2. User requests to edit certain fields of specific student in the list
+3. TAB updates the corresponding fields of the student
 
-1.  User requests to list students
-2.  TAB shows a list of students
-3.  User requests to add a lesson for a specific student in the list
-4.  TAB adds the lesson to the student specified
-    
-    Use case ends.
+   Use case ends.
 
 **Extensions**
 
-* 2a. The list is empty.
-  
+* 1a. The list is empty
+
   Use case ends.
 
-* 3a. The given index is invalid.
-  * 3a1. TAB shows an error message.
+* 2a. The given index is invalid
 
-* 3b. The specified end time is earlier than the start time
-  * 3b1. TAB shows an error message.
-    
-* 3c. There is an existing lesson for that student with the same lesson details.
-  * 3c1. TAB shows an error message. 
-    
-  Use case resumes at step 2.
+    * 2a1. TAB displays an error message
 
-**Use case: UC5 – Delete Student’s Lesson Information**
+      Use case resumes at step 1.
 
-**MSS**
+* 2b. No fields to be edited are specified
 
-1. User requests to list students. 
-   
-2. TAB shows the list of students.
-
-3. User requests to delete a lesson for the student.
+    * 2b1. TAB shows an error message
 
-4. TAB deletes the lesson from the specified student in the list.
+      Use case resumes at step 1.
 
-    Use case ends.
+* 2c. TAB detects that the entered student data does not satisfy requirements
 
-**Extensions**
+    * 2c1. TAB displays an error message and the recommended changes
 
-* 2a. The list is empty.
+      Use case resumes at step 1.
 
-  Use case ends.
+* 2d. TAB detects a duplicate student
 
-* 3a. The given index of the person is invalid.
-  * 3a1. TAB shows an error message.
+    * 2d1. TAB shows an error message
 
-* 3b. The given index of the lesson of the person is invalid.
-  * 3b1. TAB shows an error message.
+      Use case ends.
 
-  Use case resumes from step 2.
+<br/>
 
-**Use case: UC6 – Edit Student’s Lesson Information**
+####**Use case: UC4 - Delete a Student**
```

Comment:
Is there supposed to be a space between "####"? Also, for this use case, there's the use of ** ... ** but the others don't seem to have this?